examples/simple_rpc.rs

```diff
@@ -0,0 +1,102 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+/*!
+This example illustrates how client code can use the _Communication Level API_ to invoke a
+service operation. It also shows how the corresponding service provider can be implemented.
+ */
+use std::sync::Arc;
+
+use protobuf::well_known_types::wrappers::StringValue;
+use up_rust::{
+    communication::{
+        CallOptions, InMemoryRpcClient, InMemoryRpcServer, RequestHandler, RpcClient, RpcServer,
+        ServiceInvocationError, UPayload,
+    },
+    local_transport::LocalTransport,
+    LocalUriProvider,
+};
+
+struct EchoOperation {}
+
+#[async_trait::async_trait]
+impl RequestHandler for EchoOperation {
+    async fn invoke_method(
```

Comment:
I'm thinking about whether it would be more understandable if we had another function name here. For example, `reply_method` or something similar. `invoke_method` confuses me a little why the server side needs this.

Reply:
Hmm, given that the trait is called `RequestHandler`, we might want to call the function `handle` or `handle_request`? WDYT?

Reply:
Agree. I love `handle_request`.

Reply:
Then I'll create a separate PR for changing the trait